src/Raven.Server/Web/System/AdminDatabasesHandler.cs

```diff
@@ -1202,19 +1202,81 @@ private async Task<Size> CalculateStorageSize(string databaseName)
         {
             var database = GetStringQueryString("name");
+            var validate = GetBoolValueQueryString("validate", required: false);
+
             await ServerStore.EnsureNotPassiveAsync();
 
             using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
             using (var json = await context.ReadForDiskAsync(RequestBodyStream(), "unused-databases-ids"))
             {
                 var parameters = JsonDeserializationServer.Parameters.UnusedDatabaseParameters(json);
+                if (validate.HasValue && validate.Value)
+                {
+                    var token = CreateHttpRequestBoundTimeLimitedOperationToken(ServerStore.Configuration.Cluster.OperationTimeout.AsTimeSpan);
+                    await ValidateUnusedIdsAsync(parameters.DatabaseIds, database, token.Token);
+                }
+
                 var command = new UpdateUnusedDatabaseIdsCommand(database, parameters.DatabaseIds, GetRaftRequestIdFromQuery());
                 await ServerStore.SendToLeaderAsync(command);
             }
 
             NoContentStatus();
         }
 
+        private async Task ValidateUnusedIdsAsync(IEnumerable<string> unusedIds, string databaseName, CancellationToken token)
+        {
+            DatabaseTopology topology;
+            ClusterTopology clusterTopology;
+            using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            using (context.OpenReadTransaction())
+            using (var rawRecord = ServerStore.Cluster.ReadRawDatabaseRecord(context, databaseName))
+            {
+                topology = rawRecord.Topology;
+                clusterTopology = ServerStore.GetClusterTopology(context);
+            }
+
+            foreach (var id in unusedIds)
+            {
+                if (id == topology.DatabaseTopologyIdBase64)
+                {
+                    throw new InvalidOperationException($"'DatabaseTopologyIdBase64' ({id}) cannot be added to the 'unused ids' list (of '{databaseName}').");
+                }
+            
+                if (id == topology.ClusterTransactionIdBase64)
+                {
+                    throw new InvalidOperationException($"'ClusterTransactionIdBase64' ({id}) cannot be added to the 'unused ids' list (of '{databaseName}').");
+                }
+            }
+
+            var nodesUrls = clusterTopology.AllNodes
+                .Where((kvp) => topology.AllNodes.Contains(kvp.Key))
+                .Select(kvp => kvp.Value)
+            .ToArray();
+
+            using var requestExecutor = RequestExecutor.Create(nodesUrls, databaseName, Server.Certificate.Certificate, DocumentConventions.Default);
+            // ClusterRequestExecutor.Create(nodesUrls, Server.Certificate.Certificate, DocumentConventions.Default);
+
+            foreach (var nodeTag in topology.AllNodes)
+            {
+                foreach (var id in unusedIds)
```

Comment:
this make no sense at all... why to do a request for every id?

Reply:
Because I want to be sure the user doesn't add an id of a db instance to the unusedIds list.
I didn't found another wat to get the database ids from all the database nodes, also asked Aviv about that.

Reply:
Done